spec/models/response_spec.rb

```diff
@@ -103,4 +103,43 @@
       end
     end
   end
+
+  context "destroying a response" do
+    let(:question_types) do
+      ["text", {repeating: {items: ["text", {repeating: {items: %w[text select_multiple image]}}]}}]
+    end
+    let(:outer_form_grp) { form.c[1] }
+    let(:inner_form_grp) { outer_form_grp.c[1] }
+    let(:answer_values) do # for original response
+      [
+        "A",
+        {repeating: [
+          [
+            "B",
+            {repeating: [["C", %w[Dog], create(:media_image)]]}
+          ],
+          [
+            "D",
+            {repeating: [["E", %w[Cat Dog], create(:media_image)], ["F", %w[Cat], create(:media_image)]]}
+          ]
+        ]}
+      ]
+    end
+    let(:user) { create(:user) }
+    let(:form) { create(:form, question_types: question_types) }
+    let(:response) { create(:response, user: user, form: form, answer_values: answer_values) }
+
+    it "destroys nested response tree nodes, media, choices" do
+      expect(Choice.all.count).to eq 0
+      expect(Media::Image.all.count).to eq 0
+      response # create response
+      expect(Choice.all.count).to be > 0
+      expect(Media::Image.all.count).to be > 0
+      expect(Answer.where(response_id: response.id, type: "Answer").count).to eq 12
```

Comment:
`type: Answer` is redundant here b/c you're doing where on `Answer`

Reply:
But for that matter why not instead do a check of how many ResponseNodes there are before and after? Then you cover AnswerGroups, etc...

Reply:
Oh, also since the DB should be empty before each example, you can just do `ResponseNode.count` should be zero. No need to filter at all I think.